src/icrc1_auction_api.mo

```diff
@@ -185,34 +189,43 @@ actor class Icrc1AuctionAPI(trustedLedger_ : ?Principal, adminPrincipal_ : ?Prin
     };
     switch (result) {
       case (?(delta, usableBalance)) {
-        await* assetInfo.handler.trigger();
-        ignore a.setCredit(caller, assetId, Int.abs(usableBalance));
+        assert assetInfo.handler.debitUser(caller, delta);
+        ignore a.appendCredit(caller, assetId, delta);
         #Ok({ deposit_inc = delta; credit_inc = delta });
       };
-      case (null) {
-        #Ok({
-          deposit_inc = 0;
-          credit_inc = 0;
-        });
+      case (null) #Err(#NotAvailable("Deposit was not detected"));
+    };
+  };
+
+  public shared ({ caller }) func deposit_from_allowance(args : { account : ICRC1.Account; amount : Nat; token : Principal }) : async UpperResult<AccountManager.DepositFromAllowanceResult, AccountManager.DepositFromAllowanceError> {
```

Comment:
Please make this match the spec in https://github.com/research-ag/icrcX

For example function name and arguments.

Note: only subaccount is specified in the argument not the full `ICRC1.Account` type. The owning principal is always the caller.